frontend/src/components/RecordingDataGrid.tsx

```diff
@@ -14,58 +16,65 @@ const tableColumns: EuiBasicTableColumn<Recording>[] = [
     name: 'File Name',
   },
   {
-    field: 'imageName',
+    field: 'recordingImage',
     name: 'Image Name',
   },
   {
     field: 'size',
     name: 'Size',
-    render: (value: number) => prettyBytes(value, { maximumFractionDigits: 2 }),
+    render: (value: number) => prettyBytes(value, {maximumFractionDigits: 2}),
   },
   {
     field: 'date',
     name: 'Timestamp',
   },
 ]
 
-const data: Recording[] = [
-  {
-    id: 'record_1',
-    name: 'test_recording',
-    imageName: 'wheezy.qcow2',
-    date: new Date(),
-    size: 150*1024*1024,
-  },
-  {
-    id: 'record_2',
-    name: 'test_recording2',
-    imageName: 'wheezy.qcow2',
-    date: new Date(),
-    size: 150*1024*1024,
-  }
-];
-
 function RecordingDataGrid() {
   const navigate = useNavigate();
+  const location = useLocation();
+  const {isLoading, error, data} = useFindAllRecordings();
+  const queryClient = useQueryClient();
+  const deleteFunction = useDeleteRecordingById({mutation: {onSuccess: () => queryClient.invalidateQueries()}});
 
-  const getRowProps = (item: Recording) => {
-    const { id } = item;
+  const deleteRecording = ({recordingId}: {recordingId: string}) => {
+    deleteFunction.mutate({recordingId: recordingId});
+  }
+
+  useEffect(() => {
+    if(location.state) {
+      deleteRecording({recordingId: location.state.recordingId});
+    }
+  }, []);
+
+  const getRowProps: EuiBasicTableProps<Recording>['rowProps'] = (item) => {
+    const {id} = item;
     return {
       'data-test-subj': `recording-row-${id}`,
       onClick: () => {
-        navigate('/recordingDetails', {state:{item: item}});
+        navigate('/recordingDetails', {state: {item: item}});
```

Comment:
Be very wary of using React Router state. It's a much better idea to pass in the ID in the URL and let the page in question use the API hooks to fetch the data. I've run into horrific bugs using this pattern in the past, so I highly recommend you change this.

The path for the page can look something like `/recording/:id`.

Reply:
Since it would have to be changed in several other places, would it be possible to approve the PR for now and make this part of the UI cleanup task?

Reply:
Makes sense, plus we have more important things to worry about rn